source/Connection/AMDSClientTCPSocket.cpp

```diff
@@ -19,6 +19,11 @@ AMDSClientTCPSocket::AMDSClientTCPSocket(const QString host, const quint16 port,
 	socketKey_ = "";
 	tcpSocket_ = new QTcpSocket(this);
 
+	waitingMorePackages_ = false;
+	readedBufferSize_ = 0;
+	expectedBufferSize_ = 0;
+	incomeDataBuffer_ = new QByteArray();
```

Comment:
For `QByteArray`, we don't need to release any memory. It is handled by QT. For a safe handle, I will add a `incomeDataBuffer_->clear()` in the destructor function.

In the documentation, it mentioned a function called `squeeze()`, which will release unused memory when necessary. But it also mentioned `The sole purpose of this function is to provide a means of fine tuning QByteArray's memory usage. In general, you will rarely ever need to call this function.`.